src/test/java/io/github/jhipster/CoverageTest.java

```diff
@@ -0,0 +1,67 @@
+package io.github.jhipster;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.reflect.ClassPath;
+
+public class CoverageTest {
+
+    private final Logger log = LoggerFactory.getLogger(CoverageTest.class);
+
+    @Test
+    public void testCoverage() throws Exception {
+        Class<?> root = this.getClass();
+        ClassLoader loader = root.getClassLoader();
+        ClassPath path = ClassPath.from(loader);
+
+        Map<String, Set<String>> map = new HashMap<>();
+        for (ClassPath.ClassInfo info : path.getTopLevelClassesRecursive(root.getPackage().getName())) {
+            Class<?> clazz = Class.forName(info.getName(), false, loader);
+            if (clazz.isInterface()) {
+                continue;
+            }
+            String pack = info.getPackageName();
+            Set<String> set = map.get(pack);
+            if (set == null) {
+                set = new HashSet<>();
+                map.put(pack, set);
+            }
+            set.add(info.getSimpleName());
+        }
+
+        Set<String> missing = new TreeSet<>();
+        for (Map.Entry<String, Set<String>> entry : map.entrySet()) {
+            String pack = entry.getKey();
+            Set<String> classes = entry.getValue();
+            for (String clazz : classes) {
+                if (clazz.endsWith("Test")) {
+                    continue;
+                }
+                String test = clazz + "Test";
+                if (!classes.contains(test)) {
+                    missing.add(pack + "." + test);
+                }
+            }
+        }
+
+        if (missing.size() > 0) {
+            StringBuffer buffer = new StringBuffer(64 + 32 * missing.size());
+            buffer.append("Missing " + missing.size() + " classes:" + System.lineSeparator());
+            for (String clazz : missing) {
+                buffer.append("  " + clazz + System.lineSeparator());
+            }
+            log.warn(buffer.toString());
+        }
+        // TODO uncomment:
+        // assertThat(missing).isEmpty();
```

Comment:
??

Reply:
Once coverage is complete we could uncomment this, it’ll throw a failure when new classes are added but an accompanying test class is missing.

But like I said above, perhaps this is too pedantic.